src/main/kotlin/com/criptext/mail/utils/generaldatasource/data/UserDataWriter.kt

```diff
@@ -21,22 +19,73 @@ class UserDataWriter(private val emailDao: EmailDao,
 
             val tmpFileLinkData = createTempFile()
 
-            addContactsToFile(contactDao.getAll(), tmpFileLinkData)
-            addLabelsToFile(labelDao.getAll(), tmpFileLinkData)
-            addFilesToFile(fileDao.getAll(), tmpFileLinkData)
-            addMailsToFile(emailDao.getAll(), tmpFileLinkData)
-            addMailsAndLabelsRelationToFile(emailLabelDao.getAll(), tmpFileLinkData)
-            addMailsAndContactsRelationToFile(emailContactDao.getAll(), tmpFileLinkData)
-            addMailsFileKey(fileKeyDao.getAll(), tmpFileLinkData)
-            addExternalSessionsToFile(emailExternalSessionDao.getAll(), tmpFileLinkData)
-
+            addContactsToFile(db.contactDao().getAll(), tmpFileLinkData)
+            addLabelsToFile(db.labelDao().getAll(), tmpFileLinkData)
+            addMailsToFile(db.emailDao().getAllForLinkFile(), tmpFileLinkData)
+            addFilesToFile(db.fileDao().getAllForLinkFile(), tmpFileLinkData)
+            addMailsAndLabelsRelationToFile(db.emailLabelDao().getAllForLinkFile(), tmpFileLinkData)
+            addMailsAndContactsRelationToFile(db.emailContactDao().getAllForLinkFile(), tmpFileLinkData)
+            addMailsFileKey(db.fileKeyDao().getAll(), tmpFileLinkData)
 
             return tmpFileLinkData.absolutePath
         }catch (ex:Exception){
             return null
         }
     }
 
+    fun createDBFromFile(file: File) {
+        val contactWriter = ContactDataWriter(db.contactDao())
+        val labelWriter = LabelDataWriter(db.labelDao())
+        val fileWriter = FileDataWriter(db.fileDao())
+        val emailWriter = EmailDataWriter(db.emailDao())
+        val emailLabelWriter = EmailLabelDataWriter(db.emailLabelDao())
+        val emailContactWriter = EmailContactDataWriter(db.emailContactDao())
+        val fileKeyWriter = FileKeyDataWriter(db.fileKeyDao())
+        val data = file.bufferedReader()
+        var line = data.readLine()
+        db.beginTransaction()
+        while(line != null) {
+            val json = JSONObject(line)
+            when (json.getString("table")) {
+                "contact" -> contactWriter.insert(json.get("object").toString())
+                "label" -> labelWriter.insert(json.get("object").toString())
+                "email" -> emailWriter.insert(json.get("object").toString())
+                "file" -> {
+                    emailWriter.flush()
+                    fileWriter.insert(json.get("object").toString())
+                }
+                "email_label" -> {
+                    labelWriter.flush()
+                    emailWriter.flush()
+                    emailLabelWriter.insert(json.get("object").toString())
```

Comment:
maybe the writer class could have an extra field `dependencies` to avoid explicitly flushing. 

so you could do:

``` kotlin
val emailLabelWriter = EmailLabelDataWriter(db.emailLabelDao(), dependencies = listOf(labelWriter, emailWriter))
```